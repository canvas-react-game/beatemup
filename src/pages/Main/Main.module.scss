@use "./src/styles/variables.module.scss" as variables;

.container {
  padding: variables.$spacing_x1;
}

.infoContainer {
  display: flex;
  flex-direction: column;
  align-items: center;

  > a {
    margin-top: 60px;
  }
}

.title {
  font-size: xxx-large;
  color: white;
  margin-top: 50px;

  .titleMain {
    color: variables.$primary;
  }
}

.info {
  font-size: 24px;
  color: white;
  max-width: 648px;
  text-align: start;
  margin-top: 50px;
}

.gameHref:hover {
  background: darken(variables.$primary, 10%);
  color: white;
}
