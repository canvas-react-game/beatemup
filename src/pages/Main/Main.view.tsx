import React, { FC } from "react"
import { Typography } from "antd"

import Container from "@/components/Container"
import Header from "@/components/Header"

import styles from "./Main.module.scss"

const Main: FC = () => {
    const {
        gameHref, title, info, infoContainer, container,
    } = styles
    return (
        <Container>
            <Header/>
            <div className={container}>
                <div className={infoContainer}>
                    <Typography className={title}>Игра</Typography>
                    <Typography className={info}>
                        Lorem Ipsum is simply dummy text of the printing and typesetting industry.
                    </Typography>
                    <a href='' className={gameHref}>Играть</a>
                </div>
            </div>
        </Container>
    )
}

export default Main