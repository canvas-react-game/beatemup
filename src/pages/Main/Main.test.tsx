import React, { FC } from "react";
import { render } from "@testing-library/react";
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import Main from "@/pages/Main";
import { Provider } from "react-redux";
import { makeStore } from "@/helpers/tests/makeStore";
import { compose } from "@/helpers/tests/compose";

describe("Main page", () => {

    function withRouter(Component: FC): FC {
        const history = createMemoryHistory({ initialEntries: ['/'] });
        return () => (
            <Router history={history}>
                <Component />
            </Router>
        )
    }

    function withStore(Component: FC): FC {
        const store = makeStore();
        return () =>  (
            <Provider store={store}>
                <Component />
            </Provider>
        )
    }

    test("renders correctly", () => {
        const MainComposed = compose(withRouter, withStore)(Main)
        const tree = render(<MainComposed />);

        expect(tree).toMatchSnapshot();
    });

    test("has correct title", () => {
        const MainComposed = compose(withRouter, withStore)(Main)

        const title = "Игра";
        const main = render(<MainComposed />);

        expect(main.findAllByText(title)).toBeTruthy();
    });
});
