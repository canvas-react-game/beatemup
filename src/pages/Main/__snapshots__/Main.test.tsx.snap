// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="container"
        >
          <div
            class="logo"
          >
            <a
              href="/main"
            >
              Logo
            </a>
          </div>
          <div
            class="routesContainer"
          >
            <nav
              class="ulContainer"
            >
              <ul
                class="ul"
              >
                <li
                  class="inactive"
                >
                  <a
                    href="/about"
                  >
                    Об игре
                  </a>
                </li>
              </ul>
            </nav>
            <div
              class="buttonContainer"
            />
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span>
                Темная
              </span>
            </button>
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="infoContainer"
          >
            <article
              class="ant-typography title"
            >
              Игра
            </article>
            <article
              class="ant-typography info"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry.
            </article>
            <a
              class="gameHref"
              href="/game"
            >
              Играть
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="container"
      >
        <div
          class="logo"
        >
          <a
            href="/main"
          >
            Logo
          </a>
        </div>
        <div
          class="routesContainer"
        >
          <nav
            class="ulContainer"
          >
            <ul
              class="ul"
            >
              <li
                class="inactive"
              >
                <a
                  href="/about"
                >
                  Об игре
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="buttonContainer"
          />
          <button
            class="ant-btn ant-btn-primary"
            type="button"
          >
            <span>
              Темная
            </span>
          </button>
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="infoContainer"
        >
          <article
            class="ant-typography title"
          >
            Игра
          </article>
          <article
            class="ant-typography info"
          >
            Lorem Ipsum is simply dummy text of the printing and typesetting industry.
          </article>
          <a
            class="gameHref"
            href="/game"
          >
            Играть
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
